src/tests/test_fastod.cpp

```diff
@@ -0,0 +1,509 @@
+                                                      "{1,4,5,6,7} : 9>= ~ 3<=",
+                                                      "{1,4,5,6,9} : 3>= ~ 7<=",
+                                                      "{1,5,6,7,9} : 4>= ~ 3<=",
+                                                      "{1,4,5,6,7} : 3>= ~ 9<=",
+                                                      "{4,5,7,8,9} : 3>= ~ 6<=",
+                                                      "{4,5,7,8,9} : 6>= ~ 3<=",
+                                                      "{5,6,7,8,9} : 3>= ~ 4<=",
+                                                      "{5,6,7,8,9} : 4>= ~ 3<=",
+                                                      "{1,5,6,8,9} : 10>= ~ 4<=",
+                                                      "{1,5,6,8,9} : 4>= ~ 10<=",
+                                                      "{1,5,6,8,9} : 3>= ~ 4<=",
+                                                      "{1,5,6,8,9} : 4>= ~ 3<=",
+                                                      "{1,4,5,7,8} : 9>= ~ 6<=",
+                                                      "{1,4,5,7,8} : 6>= ~ 9<=",
+                                                      "{4,6,7,8,9,10} : 5>= ~ 3<=",
+                                                      "{4,6,7,8,9,10} : 3>= ~ 5<=",
+                                                      "{2,5,6,7,8,9,10} : 1>= ~ 4<=",
+                                                      "{2,5,6,7,8,9,10} : 4>= ~ 1<="};
+
+    std::vector<std::string> expected_simple_ods_str = {
+            "{3} : [] -> 2<=",           "{3,6} : [] -> 9<=",  "{3,6} : [] -> 7<=",
+            "{3,9} : [] -> 5<=",         "{3,5} : [] -> 9<=",  "{3,10} : [] -> 4<=",
+            "{3,7} : [] -> 1<=",         "{3,8} : [] -> 7<=",  "{3,7} : [] -> 8<=",
+            "{3,8} : [] -> 1<=",         "{3,4} : [] -> 1<=",  "{3,8} : [] -> 4<=",
+            "{3,4} : [] -> 8<=",         "{3,10} : [] -> 7<=", "{3,5} : [] -> 4<=",
+            "{3,4} : [] -> 5<=",         "{3,10} : [] -> 9<=", "{3,7} : [] -> 4<=",
+            "{3,4} : [] -> 7<=",         "{3,9} : [] -> 4<=",  "{3,4} : [] -> 9<=",
+            "{3,6} : [] -> 4<=",         "{3,10} : [] -> 8<=", "{3,10} : [] -> 1<=",
+            "{3,5} : [] -> 1<=",         "{3,8} : [] -> 5<=",  "{3,5} : [] -> 8<=",
+            "{3,9} : [] -> 1<=",         "{3,9} : [] -> 8<=",  "{3,8} : [] -> 9<=",
+            "{3,7} : [] -> 5<=",         "{3,5} : [] -> 7<=",  "{3,6} : [] -> 1<=",
+            "{3,6} : [] -> 8<=",         "{3,10} : [] -> 5<=", "{3,6} : [] -> 5<=",
+            "{3,9} : [] -> 7<=",         "{3,7} : [] -> 9<=",  "{1,4,5,6,8} : [] -> 9<=",
+            "{1,4,5,6,7,10} : [] -> 9<="};
+
+    TestFastod(dataset_path, std::move(expected_asc_ods_str), std::move(expected_desc_ods_str),
+               std::move(expected_simple_ods_str));
+}
+
+using DatasetPathAndHash = std::tuple<std::string, std::string, size_t>;
```

Comment:
use `TableConfigHash` from `src/tests/table_config.h` after fixing the comment above (about csv configs)

Reply:
File `src/tests/table_config.h` not exists. I found struct `CSVConfigHash` in `src/tests/csv_config_util.h` and used it.